### DISCUSSION TITLE
After ngx update to 2.11.6: NLTK-Error on every new document

### DISCUSSION THREAD

**Discussion by schweiger2** [tteck/Proxmox#3596] [Help Needed] [answered]
Hey y'all,

after the last update to 2.11.6 all OCR-recognitions finish with this error:

Resource �[93mpunkt_tab�[0m not found. Please use the NLTK Downloader to obtain the resource: �[31m>>> import nltk >>> nltk.download('punkt_tab') �[0m For more information see: https://www.nltk.org/data.html Attempted to load �[93mtokenizers/punkt_tab/german/�[0m Searched in: - PosixPath('/usr/share/nltk_data')

Because its a bare metal installation by script and updates are performed by scripts too an updated update-script would be appreciated.

Please help a noob. Thanks a lot.

**Comment by tornenen** [comment]
Just go to https://www.nltk.org/nltk_data/ download and unzip the packages described here on point 15: https://docs.paperless-ngx.com/setup/#bare_metal and upload them to /usr/share/nltk_data in the right directory.
Restart paperless and everything is fine again.

**Reply by schweiger2** [reply]
So I just download all files with Snowball, Stemmer, Stopwords and Punkt tokenizer in their names and unzip them to the appropriate folder ?

Why does NLTK talk about subfolders named chunkers, grammars, misc, sentiment, taggers, corpora, help, models, stemmers, tokenizers ?

On first sight that may not fit ?

**Reply by tornenen** [reply]
I think you need to put the "punkt" in "tokenizers", the "snowball" in "stemmers" and "stopwords" in "corpora". 
Thats how the folders looked before my update. So after the update i did extract them in these folders and everything worked as expected.

**Reply by schweiger2** [reply, marked as answer]
What a surprise. 

I found that complete NLTK Data was in the folders before downloading, except:
- snowball_data was zipped - I unzipped it
- punkt was there but punkt_tab was missing - I downloaded and unzipped

Watch out, after unzipping you need to move the folder in the first folder to the appropriate NLTK-folders.

Restart - Working

Thanks for your hints and help.

**Reply by sierralima123** [reply]
To fix this issue with NLTK, follow this procedure:
```
cd /usr/share/nltk_data
python3 -m nltk.downloader punkt_tab
```
Please check your error message and replace "punkt_tab" with your missing module.

**Comment by HackerJL** [comment]
If I can add my 2 cents that no one asked for.  The update command for ngx could have a third option to re-download these and put them in their places, like the script does when it first sets it up.  Or maybe even an option to do that as part of the update.  I just moved from 2.11.4 to 2.11.6 tonight and everything seems fine, no additional steps were needed (this time?)

**Comment by tteck** [comment]
https://github.com/tteck/Proxmox/issues/3643
